sogs/routes/users.py

```diff
@@ -76,16 +76,89 @@ def extract_rooms_or_global(req, admin=True):
     return (None, True)
 
 
-def _serialize_message(msg):
+def _serialize_message(msg, *, key):
     return {
         "id": msg.id,
         "posted_at": msg.posted_at,
         "expires_at": msg.expires_at,
         "message": utils.encode_base64(msg.data),
-        "sender": msg.sender.session_id,
+        key: getattr(msg, key).session_id,
     }
 
 
+def _box(func, **kwargs):
+    """ handle inbox/outbox endpoints common logic """
+    if g.user.banned or not g.user.is_blinded:
+        abort(http.FORBIDDEN)
+    keys = {'to': 'sender', 'sent': 'recipient'}
+    key = keys.get(func)
+    limit = utils.get_int_param('limit', 100, min=1, max=256, truncate=True)
+    get = getattr(Message, func)
+    msgs = [_serialize_message(msg, key=key) for msg in get(user=g.user, limit=limit, **kwargs)]
+    if msgs or not kwargs:
+        return jsonify(msgs)
+    return Response('', status=http.NOT_MODIFIED)
+
+
+@users.get("/outbox")
+@auth.user_required
+def get_outbox():
+    """
+    Retrieves all of the messages we have sent to others via their inbox (up to `limit`).
+
+    # Query Parameters
+
+    The request takes an optional `limit` query parameter indicating the number of messages to
+    return (up to 256).  If omitted, at most 100 messages are returned.
+
+    # Return value
+
+    Returns a JSON array of up to `limit` (default 100) messages, with oldest messages first.  Each
+    element is a JSON object with keys:
+
+    - `id` — the unique integer message id.
+    - `posted_at` — unix timestamp (float) when the message was received by SOGS.
+    - `expires_at` — unix timestamp (float) when SOGS will expire and delete the message.
+    - `message` — the encrypted message body.
+    - `recipient` — the (blinded) Session ID of the recipient of the message.
+
+    # Encryption details
+
+    see [/inbox](#GET-inbox)
+
+    """
+    return _box("sent")
+
+
+@users.get("/outbox/since/<int:msgid>")
+@auth.user_required
+def poll_outbox(msgid):
+    """
+    Polls for any DMs sent since the given id.
+
+    # URL Parameters
+
+    - `msgid` — the numeric message ID of the last sent message.  (Newer messages will always
+      have a higher ID).
+
+    # Query Parameters
+
+    The request takes an optional `limit` query parameter indicating the number of messages to
+    return (up to 256).  If omitted, at most 100 messages are returned.
+
+    # Return value
+
+    This method, on success, returns *either* a 200 (OK) status code with a list of 1 or more new
+    messages, or else returns a 304 (Not Modified) response with an empty body to indicate that
+    there are no messages for this user newer than the given ID.
+
+    If there are messages this returns a JSON array of up to `limit` (default 100) messages, with
+    oldest messages first.  Each element is exactly as described in the [all messages](#GET-outbox)
+    endpoint.
+    """
```

Comment:
I feel like we can substantially dry out these docstrings by extracting all the common bits into globals and using f-strings for the docstring.

Reply:
(Just make sure the global has the same 4-space indent, so that it gets properly stripped off when the docstring gets cleaned up).